src/test/java/org/broadinstitute/hellbender/tools/copynumber/utils/segmentation/KernelSegmenterUnitTest.java

```diff
@@ -0,0 +1,126 @@
+package org.broadinstitute.hellbender.tools.copynumber.utils.segmentation;
+
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class KernelSegmenterUnitTest extends BaseTest {
+    private static final int RANDOM_SEED = 1;   //reset seed before each simulated test case
+
+    @DataProvider(name = "dataKernelSegmenter")
```

Comment:
Could you briefly describe the two simulated test data in the unit test class javadoc (e.g. the first case is Gaussian data with discrete jumps in the mean every 100 points; the second case simulates a multimodal series such that the mean alternates in sign successively while the absolute value of the mean jumps every 100 points).

I would also include a few corner cases (empty data, empty window size, single-segment simulated data), and perhaps an example for a small but obvious changepoint + long wavelength sinusoidal background and how including small window sizes help detect it.

Reply:
Good points.  Added everything except the sinusoidal background case.  Although useful, I think it's a bit beyond the scope of the unit test.

Reply:
*cough*lazy*cough* OK! ;-)